syntax = "proto3";

package zdb.kvStore;
import "proto/types.proto";

service KVStoreService {
    rpc get(GetRequest) returns (GetReply) {}
    rpc set(SetRequest) returns (SetReply) {}
    rpc erase(EraseRequest) returns (EraseReply) {}
    rpc size(SizeRequest) returns (SizeReply) {}
}

message GetRequest {
    zdb.proto.Key key = 1;
    zdb.proto.RequestID requestID = 2;
}

message GetReply {
    zdb.proto.Value value = 1;
}

message SetRequest {
    zdb.proto.Key key = 1;
    zdb.proto.Value value = 2;
    zdb.proto.RequestID requestID = 3;
}

message SetReply {
}

message EraseRequest  {
    zdb.proto.Key key = 1;
    zdb.proto.RequestID requestID = 2;
}

message EraseReply {
    zdb.proto.Value value = 1;
}

message SizeRequest {
    zdb.proto.RequestID requestID = 2;
}

message SizeReply {
    uint64 size = 1;
}
